pkg/services/ngalert/eval/eval.go

```diff
@@ -0,0 +1,184 @@
+// Package eval executes the condition for an alert definition, evaluates the condition results, and
+// returns the alert instance states.
+package eval
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/tsdb"
+)
+
+// Condition contains backend expressions and queries and the RefID
+// of the query or expression that will be evaluated.
+type Condition struct {
+	RefID string `json:"refId"`
+
+	QueriesAndExpressions []backend.DataQuery `json:"queriesAndExpressions"`
+}
+
+// ExecutionResults contains the unevaluated results from executing
+// a condition.
+type ExecutionResults struct {
```

Comment:
Doesn't have to be public:
```suggestion
// executionResults contains the unevaluated results from executing
// a condition.
type executionResults struct {
```